pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -671,6 +691,42 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	return deployment, nil
 }
 
+func (c *Cluster) createOSDService(osd OSDInfo, labels map[string]string) (string, error) {
+	selectorLabels := map[string]string{
+		k8sutil.AppAttr: AppName,
+		"ceph-osd-id":   strconv.FormatInt(int64(osd.ID), 10),
+	}
+	svcDef := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("rook-ceph-osd-%d", osd.ID),
+			Namespace: c.clusterInfo.Namespace,
+			Labels:    labels,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: selectorLabels,
+			Ports: []v1.ServicePort{
+				{
+					Name:       "osd-port",
+					Port:       6800,
+					TargetPort: intstr.FromInt(6800),
+					Protocol:   v1.ProtocolTCP,
+				},
+			},
+		},
+	}
+	err := c.clusterInfo.OwnerInfo.SetOwnerReference(svcDef)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to set owner reference to osd service %q", svcDef.Name)
+	}
+
+	svc, err := k8sutil.CreateOrUpdateService(c.clusterInfo.Context, c.context.Clientset, c.clusterInfo.Namespace, svcDef)
+	if err != nil {
+		return "", err
+	}
+
+	return svc.Spec.ClusterIP, err
```

Comment:
Isn't there something different we needed to do for the global IP, instead of just using the cluster IP?

Reply:
fixed